src/main/java/frc/robot/subsystems/shooter/Shooterreal.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.subsystems.shooter;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+public class Shooterreal implements ShooterIO{
+private final TalonFX motor = new TalonFX(0);
+private double pivotAppliedVolts;
+
+@Override
+public ShooterIOInputs updateInputs() {
+  ShooterIOInputs inputs = new ShooterIOInputs();
+  inputs.PivotAmps = motor.getSupplyCurrent().getValue();
+  inputs.PivotTempc = motor.getDeviceTemp().getValue();
+  inputs.PivotMeters = motor.getRotorPosition().getValue();
+  inputs.PivotVoltage = motor.getSupplyVoltage().getValue();
+  inputs.PivotVelocity = motor.getVelocity().getValue();
+  inputs.FlywheelAmp = motor.getSupplyCurrent().getValue();
+  inputs.FlywheelTempc = motor.getDeviceTemp().getValue();
+  inputs.FlywheelVoltage = motor.getSupplyVoltage().getValue();
+  inputs.FlywheelVelocity = motor.getVelocity().getValue();
+  inputs.pivotAppliedVolts = pivotAppliedVolts;
+  return inputs;
+}
+
+public void setFlywheelVelocity(final Rotation2d rotation) {
+  setFlywheelVelocity(
+    pivotAppliedVolts = MathUtil.clamp(volts, -12.0, 12.0));
+    motor.setVoltage(pivotAppliedVolts);
+  }
+ 
+
+private final PIDController pivotController = new PIDController(100.0, 0.0, 0.0);
```

Comment:
you dont need a PIDController object here, just use a VelocityVoltage control mode